src/main/java/teammates/storage/api/InstructorsDb.java

```diff
@@ -96,6 +98,27 @@ public InstructorAttributes regenerateEntityKey(InstructorAttributes originalIns
         return getSearchManager().searchInstructors(queryString);
     }
 
+    /**
+     * Checks if the given instructors exist in the given course.
+     */
+    public boolean hasExistingInstructorsInCourse(String courseId, Collection<String> instructorEmailAddresses) {
+        if (instructorEmailAddresses.isEmpty()) {
+            return true;
+        }
+        Set<String> existingInstructorEmailAddressesInCourse = load().filter("courseId =", courseId)
+                .project("email")
+                .list()
+                .stream()
+                .map(Instructor::getEmail)
+                .collect(Collectors.toSet());
```

Comment:
@moziliar any idea? Also to check, does this query minimize number of reads?

Reply:
Looks like projected query is counted as small operation that is billed less. It might be faster, but require some empirical data to support the claim.

Projection basically `SELETE` the projected fields. It involves lower network bandwidth and cost (not free).

Reply:
On another thought... this is probably still 1 entity read; with student, that's two.

Per FeedbackSession request, we are adding 2 reads. Can we do it only when necessary?

https://cloud.google.com/datastore/pricing Small Operations

Reply:
>To check: Do we need an index for this? Not familiar with projection queries.

From my understanding, each property of an entity is already in the [built-in indices](https://cloud.google.com/datastore/docs/concepts/indexes#:~:text=By%20default%2C%20a%20Datastore%20mode%20database%20automatically%20predefines%20an%20index%20for%20each%20property%20of%20each%20entity%20kind.%20These%20single%20property%20indexes%20are%20suitable%20for%20simple%20types%20of%20queries.). Correct me if I'm wrong though, I am still quite new to Datastore.

>Also to check, does this query minimize number of reads?

>this is probably still 1 entity read; with student, that's two.

Yes, this is my understanding as well. If you get all the email addresses, that is one single query, not for each email address but for all the email addresses, so the number of queries is always one no matter the number of email addresses. So it reduces it if you have multiple email addresses to check, but yea, I did not consider the fact that this also means that even if the deadline maps do not change, there would still be some extra unnecessary reads regardless.

>Can we do it only when necessary?

I think it should be possible.

Reply:
The query is ok as the E2E is passing.

> I think it should be possible.

Just gotta fix up this part. Since the update to a FeedbackSession and its deadlines are mutually exclusive, we can hide this away from the FeedbackSession Edit data path entirely.